src/Orleans.Runtime/Lifecycle/SiloLifecycleStage.cs

```diff
@@ -0,0 +1,29 @@
+﻿
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// Stages of a silo's lifecycle.
+    /// </summary>
+    public enum SiloLifecycleStage
+    {
+        /// <summary>
+        /// Initialize silo runtime
+        /// </summary>
+        RuntimeInitialize = 1 << 10,
+
+        /// <summary>
+        /// Start runtime services
+        /// </summary>
+        RuntimeServices = RuntimeInitialize + 1 << 10,
+
+        /// <summary>
+        /// Start application layer services
+        /// </summary>
+        ApplicationServices = RuntimeServices + 1 << 10,
+
+        /// <summary>
+        /// Silo is active and available to service requests
+        /// </summary>
+        SiloActive = ApplicationServices + 1 << 10,
```

Comment:
Would you mind using parenthesis? I believe the operator precedence is not what you intend, but even if it is, can you make it explicit to avoid confusion?